src/components/Charts/StarXYChart.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <div
+    class="flex flex-col justify-center items-center select-none"
+    :class="classname"
+  >
+    <svg ref="svgElRef" @click="handleSVGElementClick"></svg>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, onMounted, onUpdated, ref } from "vue";
+// For customizing multi mode chart: base on create date or timeline, we have rewrited the chart.xkcd's XY chart with TypeScript.
+// Here are some reasons about this motivation.
+//
+// The shortcomings of the old chart.xkcd (project link: https://github.com/timqian/chart.xkcd):
+// 1. The X axis label string could be only formatted date or raw text;
+// 2. It's difficult/not possible to debug and customize;
+//
+// The advantages of the new one:
+// 1. Writing in TypeScript originally;
```

Comment:
1. Writing in native TypeScript;
2. Easy to debug chart internal;
3. Totally customizable.